pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -219,10 +178,14 @@ func TestCreatePod(t *testing.T) {
 	for _, item := range testCases {
 		ctx, cancel := context.WithCancel(context.Background())
 		fakeClientset := fake.NewSimpleClientset()
-		controller := NewNoExecuteTaintManager(ctx, fakeClientset, (&podHolder{pod: item.pod}).getPod, getNodeFromClientset(fakeClientset), getPodsAssignedToNode(fakeClientset))
+		podIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, nil)
+		nodeIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, nil)
+		controller := setupNewNoExecuteTaintManager(ctx, fakeClientset, podIndexer, nodeIndexer)
 		controller.recorder = testutil.NewFakeRecorder()
 		go controller.Run(ctx)
 		controller.taintedNodes = item.taintedNodes
+
+		podIndexer.Add(item.pod)
 		controller.PodUpdated(nil, item.pod)
 		// wait a bit
 		time.Sleep(timeForControllerToProgress)
```

Comment:
For a follow up (not required for your enhancement): replace this with `wait.Until`